NaCl.net/Internal/StreamSalsa20Xor.cs

```diff
@@ -58,15 +51,13 @@ internal static class StreamSalsa20Xor
 
             if (mlen != 0)
             {
-                Salsa20.Transform(block, input, kcopy, ReadOnlySpan<byte>.Empty);
+                Salsa20.Transform(block, input, kcopy, null);
                 for (int i = 0; i < mlen; i++)
                 {
                     c[i] = (byte) (m[i] ^ block[i]);
                 }
             }
-
-            block.Clear();
-            kcopy.Clear();
```

Comment:
This is important, to not leave the private data in memory

Reply:
I will have a look, I have been trying so many things to try and improve performance :)

Reply:
How would you feel if we keep the data on the heap and the "memset" set heap before leaving.

Reply:
The unit test is a stupid mistake

Reply:
Worth noting that the libsodium variant would provide for `sodium_malloc`, which also places the buffers in a guarded heap, and automatically zeros the memory via `sodium_free`, with the added benefit of locking the memory region and being somewhat page resistant (OS level paging tweaks still required for 100% certainty)